packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -318,12 +369,34 @@ export class OdspDocumentService implements IDocumentService {
                 this.epochTracker,
                 requestSocketToken,
                 options,
+                refreshJoinSessionPolicy,
                 this.hostPolicy.sessionOptions?.unauthenticatedUserDisplayName,
             );
 
-        // Note: The sessionCache is configured with a sliding expiry of 1 hour,
-        // so if we've fetched the join session within the last hour we won't run executeFetch again now.
-        return this.cache.sessionJoinCache.addOrGet(this.joinSessionKey, executeFetch);
+        let response: ISocketStorageDiscovery;
+        const currentTime = Date.now();
+        // If the join session refresh time is in future, that means we can reuse the cached response. Otherwise
+        // we need to clear the cache and execute the network call, so as to refresh the session on the relay
+        // ordering service.
+        if (this.joinSessionRefreshTime === undefined || this.joinSessionRefreshTime <= currentTime) {
```

Comment:
I think (correct me if I got it wrong) that join session results in cache are shared across interactive and summarizer containers. If so, it would be great for expiration to be stored in cache and all containers / driver instances to be able to reason about.

if that's not the case and we do not share it - it's interesting question - should we? It feels like it's useful, and helps with page reload as well.